.github/workflows/ami-cleanup-ci.yaml

```diff
@@ -0,0 +1,43 @@
+name: AMI cleanup tool CI
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  cancel-in-progress: true
+  group: "Only run one instance of AMI cleanup CI for PR #${{ github.event.number }}"
+
+jobs:
+  check-if-should-run:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      should-verify-pr: ${{ steps.filter.outputs.changed }} # True if the AMI cleanup tool changed, false otherwise
+    steps:
+      - name: Filter out unrelated changes
+        uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        id: filter
+        with:
+          filters: |
+            changed: "tools/ami-cleanup/**"
```

Comment:
I _think_ this will be caught, but I don't have an easy way to test in this PR. There is a symlink in the workflows directory which (hopefully) renovate will follow.